app/src/main/java/org/thoughtcrime/securesms/onboarding/LinkDeviceActivity.kt

```diff
@@ -1,230 +1,160 @@
+            HorizontalPager(
+                state = pagerState,
+                modifier = Modifier.weight(1f)
+            ) { page ->
+                val title = TITLES[page]
+
+                when (title) {
+                    R.string.sessionRecoveryPassword -> RecoveryPassword(state, onChange, onContinue)
+                    R.string.qrScan -> MaybeScanQrCode(viewModel.qrErrorsFlow, onScan = onScan)
+                }
             }
-
-            binding.loader.isVisible = false
         }
     }
-
-    private fun register(skipped: Boolean) {
-        restoreJob?.cancel()
-        binding.loader.isVisible = false
-        TextSecurePreferences.setLastConfigurationSyncTime(this, System.currentTimeMillis())
-        val intent = Intent(this@LinkDeviceActivity, if (skipped) DisplayNameActivity::class.java else PNModeActivity::class.java)
-        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
-        push(intent)
-    }
-    // endregion
 }
 
-// region Adapter
-private class LinkDeviceActivityAdapter(private val activity: LinkDeviceActivity) : FragmentPagerAdapter(activity.supportFragmentManager) {
-    val recoveryPhraseFragment = RecoveryPhraseFragment()
-
-    override fun getCount(): Int {
-        return 2
-    }
-
-    override fun getItem(index: Int): Fragment {
-        return when (index) {
-            0 -> recoveryPhraseFragment
-            1 -> {
-                val result = ScanQRCodeWrapperFragment()
-                result.delegate = activity
-                result.message = activity.getString(R.string.activity_link_device_qr_message)
-                result
-            }
-            else -> throw IllegalStateException()
+@Preview
+@Composable
+fun PreviewRecoveryPassword() = RecoveryPassword(state = LinkDeviceState())
+
+@Composable
+fun RecoveryPassword(state: LinkDeviceState, onChange: (String) -> Unit = {}, onContinue: () -> Unit = {}) {
+    Column(
+        modifier = Modifier.padding(horizontal = 60.dp)
+    ) {
+        Spacer(Modifier.weight(1f))
+        Row {
+            Text(stringResource(R.string.sessionRecoveryPassword), style = MaterialTheme.typography.h4)
+            Spacer(Modifier.width(6.dp))
+            Icon(
+                painter = painterResource(id = R.drawable.ic_shield_outline),
+                contentDescription = null,
+            )
         }
-    }
-
-    override fun getPageTitle(index: Int): CharSequence {
-        return when (index) {
-            0 -> activity.getString(R.string.activity_link_device_recovery_phrase)
-            1 -> activity.getString(R.string.activity_link_device_scan_qr_code)
-            else -> throw IllegalStateException()
+        Spacer(Modifier.size(28.dp))
+        Text(stringResource(R.string.activity_link_enter_your_recovery_password_to_load_your_account_if_you_haven_t_saved_it_you_can_find_it_in_your_app_settings))
+        Spacer(Modifier.size(24.dp))
+        SessionOutlinedTextField(
```

Comment:
I mentioned creating this in another post. Looks like it already exist but wasn't used everywhere.
Also the text in SessionOutlinedTextfield is not set to `base`